program/tests/tests/solana_assumptions.rs

```diff
@@ -258,3 +264,122 @@ async fn test_stake_accounts() {
         }
     );
 }
+
+async fn update_commission(
+    context: &mut Context,
+    vote_account: &Pubkey,
+    withdraw_authority: &Keypair,
+    new_commission: u8,
+) -> solana_sdk::transport::Result<()> {
+    send_transaction(
+        &mut context.context,
+        &mut context.nonce,
+        &[vote_instruction::update_commission(
+            &vote_account,
+            &withdraw_authority.pubkey(),
+            new_commission,
+        )],
+        vec![&withdraw_authority],
```

Comment:
In this case we pass the withdraw authority both to the `update_commission` instruction, and we use it as the signer. But what if we pass the right withdraw authority, but sign with something else, like the private key of the vote account? Is changing the commission still rejected then?

Reply:
I hope so. Added some changes 👍

Reply:
The problem is that signing panics with `KeypairPubkeyMismatch`. I don't think we need to test this assumption :P

Reply:
Hmm, ok, good to know.